subt/octomap.py

```diff
@@ -231,7 +232,7 @@ def on_octomap(self, data):
         x = self.pose3d[0][0] - self.start_xyz[0]
         y = self.pose3d[0][1] - self.start_xyz[1]
         start = int(512 + 2*x), int(512 - 2*y)
-        img = data2maplevel(data, level=1)  # 0.5m above the ground?
+        img = data2maplevel(data, level=int(round(self.zlevel/0.5)))  # 0.5m above the ground?
```

Comment:
Is it time to stop putting `0.5` at random places in the code and replace it with a single configurable parameter that is used in all the places?

My concerns is that once we say something like "actually, 0.6 is a better value", for whatever reason, it will be a pain to change the code and we will never be confident we did it correctly. Especially for Z you are already thinking about changing the resolution of the octomap.

Reply:
it sounds like it would be good to have it as config parameter or even better propagated with binary octomap data. Agree.

Reply:
Propagating it with the data sounds like a very interesting idea. Is it feasible?

Reply:
well, I plan to extend the original binary octomap by some extra "header", so that you can send only updates. It could be some specification of the "root node" (xyz?) and then sub-octotree. Also note, that octomap has (0, 0, 0) at the robot start position which makes it hard to share it with other robots -> some offsets is needed ... but to answer your question, hmm, at the moment I do not know how to nicely extend it without hardcoding in `cloudsim2osgar` :(. So I will probably go the "json config way" :(.